schema/geometry_schema.json

```diff
@@ -643,22 +842,79 @@
 						"null"
 					]
 				},
+				"window_definition_type": {
+					"type": "string",
+					"enum": [
+						"Single Window",
+						"Multiple Windows",
+						"Window to Wall Ratio"
+					]
+				},
+				"wwr": {
+					"description": "Target window to wall ratio, only used if window_definition_type is \"Window to Wall Ratio\"",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"maximum": 1,
+					"exclusiveMaximum": true
+				},
 				"height": {
-					"description": "Vertical height of each window. In drawing units.",
-					"type": "number"
+					"description": "Vertical height of each window. In drawing units. Used if window_definition_type is \"Single Window\" or \"Multiple Windows\".",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
+				},
+				"window_spacing": {
+					"description": "Spacing between windows.  Only used if window_definition_type is  \"Multiple Windows\".",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
 				},
 				"width": {
-					"description": "Horizontal width of each window. In drawing units.",
-					"type": "number"
+					"description": "Horizontal width of each window. In drawing units. Used if window_definition_type is \"Single Window\" or \"Multiple Windows\".",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
 				},
 				"sill_height": {
 					"description": "Vertical distance from the floor to the window sill. In drawing units.",
-					"type": "number"
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
+				},
+				"overhang_projection_factor": {
+					"description": "Projection factor for overhangs, set to null or 0 for no overhang.",
```

Comment:
I've never heard of an "overhang projection factor" before and had to look it up. According to the BCL measure for this "The projection factor is the overhang depth divided by the window height." Does that mean we need an overhang depth argument too?

Reply:
Should there be a fins property too?

Reply:
Yeah there should be, I left it out but know it’ll get asked for, also window spacing should be allowed for wwr option to support fins

Reply:
Don’t need overhang depth can calculate from projection factor and window height